### DISCUSSION TITLE
Difference between entityrepository.create and constructor

### DISCUSSION THREAD

**Discussion by rubiin** [mikro-orm/mikro-orm#5666] [General]
```ts
@Entity()
export class BatchConfiguration extends BaseEntity {


  @ManyToOne()
  batch?: Rel<BootCampBatch>;

  @Property()
  isPublished = false;

}
```

I have a BatchConfig as above. The below code works  for creating batch config with the schema above

```ts
    const batch = await this.getOne(dto.batch);

    const config = this.batchConfigurationRepo.create({
      ...omit(dto, ["batch"]),
      batch,
    });

    await this.batchConfigurationRepo
      .getEntityManager()
      .persistAndFlush(config);
   
 return config;

```

But if I modify the above entity code to include constructor as 

```ts
@Entity()
export class BatchConfiguration extends BaseEntity {


  @ManyToOne()
  batch?: Rel<BootCampBatch>;

  @Property()
  isPublished = false;

constructor(partial?:Partial<BatchConfiguration>){
    super();
    Object.assign(this, partial);
} 

}
```

this give me a error that _id is not defined  of `batch` when persisting with above code

![image](https://github.com/mikro-orm/mikro-orm/assets/8222059/ade8cc83-204f-4c96-b873-08de9a288e19)

**Comment by rubiin** [comment]
Am I missing something

**Comment by B4nan** [comment]
Using `Object.assign(this, partial);` in entity constructor is an antipattern to begin with, since this way you can easily put wrong values in place of relations (e.g. FK instead of entity reference). But I don't think that is your problem here.

> this give me a error that _id is not defined of batch when persisting with above code

That error is not about `_id` being undefined, it's about the `data` being undefined. This could point to wrong metadata, first thing I would suggest is making sure this is not just about outdated metadata cache. Then check what metadata you get for that entity, there should be `constructorParams` array for each entity, matching your constructor parameter names.

I would have to see a complete reproduction to tell more.

**Reply by rubiin** [reply]
sure . I will try to get a sample repro if the above methods dont work

**Reply by rubiin** [reply]
```Object.assign(this, partial);```  can i replace it with `wrap(this).assign(data)`
